source/server/connection_handler_impl.cc

```diff
@@ -247,12 +247,16 @@ void ConnectionHandlerImpl::ActiveSocket::newConnection() {
   if (new_listener != nullptr) {
     // TODO(sumukhs): Try to avoid dynamic_cast by coming up with a better interface design
     ActiveTcpListener* tcp_listener = dynamic_cast<ActiveTcpListener*>(new_listener);
-    ASSERT(tcp_listener != nullptr, "ActiveSocket listener is expected to be tcp");
+    ASSERT(tcp_listener != nullptr, "ActiveTcpSocket listener is expected to be tcp");
     // Hands off connections redirected by iptables to the listener associated with the
     // original destination address. Pass 'hand_off_restored_destination_connections' as false to
-    // prevent further redirection.
-    tcp_listener->onAccept(std::move(socket_),
-                           false /* hand_off_restored_destination_connections */);
+    // prevent further redirection as well as 'rebalanced' as false since the connection has
+    // already been balanced if applicable. Note also that we must account for the number of
+    // connections properly across both listeners.
+    // TODO(mattklein123): See note in ~ActiveTcpSocket() related to making this accounting better.
+    listener_.decNumConnections();
+    tcp_listener->incNumConnections();
+    tcp_listener->onAcceptWorker(std::move(socket_), false, true);
```

Comment:
How does it contradict the comment?

re: why are they already balanced because we balance on the initial accept which has already been done. I will make the comment more clear.

Reply:
> How does it contradict the comment?

The comment says 'rebalanced' as false?

> re: why are they already balanced because we balance on the initial accept which has already been done. I will make the comment more clear.

Reply:
I fixed the comment to say rebelanced is true in my last push...